code/renderers/vue3/src/render.ts

```diff
@@ -11,35 +11,67 @@ export const render: ArgsStoryFn<VueFramework> = (props, context) => {
       `Unable to render story ${id} as the component annotation is missing from the default export`
     );
   }
-
   return h(Component, props);
 };
 
+const appsAndArgsByDomElementKey = new Map<
+  string,
+  { app: ReturnType<typeof createApp>; reactiveArgs: any }
+>();
+
+const STORYBOOK_ROOT_ID = 'storybook-root';
+
+function teardown(domElementKey: string) {
+  if (!appsAndArgsByDomElementKey.has(domElementKey)) {
+    return;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- we know it exists because we just checked
+  appsAndArgsByDomElementKey.get(domElementKey)!.app.unmount();
+  appsAndArgsByDomElementKey.delete(domElementKey);
+}
+
 let setupFunction = (app: any) => {};
 export const setup = (fn: (app: any) => void) => {
   setupFunction = fn;
 };
 
-const map = new Map<Element, ReturnType<typeof createApp>>();
+/* eslint-disable no-param-reassign */
+function updateArgs(reactiveArgs: any, nextArgs: any) {
+  if (!nextArgs) return;
+  Object.entries(nextArgs).forEach(([key, value]) => {
+    reactiveArgs[key] = value;
+  });
+  Object.keys(reactiveArgs).forEach((key) => {
+    if (!nextArgs[key]) {
+      reactiveArgs[key] = undefined;
+    }
+  });
```

Comment:
this basically mutates the existing - reactive - args with the new incoming args. afterwards it ensures that if an arg has been unset, it will be passed as `undefined`.

I hope that there is a more elegant way to achieve this, as this doesn't look that readable to me.